### PR TITLE
Add ony_smart_detections param

### PR DESCRIPTION
**Pull request by J3n50m4t**
Added a param to only upload smart Detections of Unifi G4 models. 

I've 15 G4 cameras and personally only care about Smart Detections, so I made this PR.

### CODE REVIEW

**Comment by ep1cman** [comment]
Hi, thanks for the pull request. I like the concept and agree it should be a setting. 

I think it would be great if instead of a boolean just for smart dectections it was instead a list of what types to backup. So `motion` `person` and `vehicle` as options, defaulting to all of them.

**Comment by J3n50m4t** [comment]
Great idea. Didnt think of it. 
I can try to adjust the PR, but i'm kind of rusty in python, so I need some time :)

**Comment by ep1cman** [comment]
More than happy to review code and help where needed! I didn't want to just come along and do it for you incase you wanted to give it a go.

File: unifi_protect_backup/unifi_protect_backup.py [hunk 1]
```diff
@@ -419,6 +423,8 @@ def _websocket_callback(self, msg: WSSubscriptionMessage) -> None:
             return
         if msg.new_obj.end is None:
             return
+        if self.only_smart_detections and (" ".join(msg.new_obj.smart_detect_types)) not in self.only_smart_detections:
```
Comment (J3n50m4t) [inline unifi_protect_backup/unifi_protect_backup.py:426]: I'm not quite happy with this line but I've no idea how to do it "correctly"

**Comment by ep1cman** [comment]
I have made some improvements. What do you think?

**Review by J3n50m4t — COMMENTED** [review]
LGTM
Love the idea of switching my version to general detection types. Even more flexibility to get motion and person but no vehicle. This would allow G3 Cameras and G4 Only Smart features.  
Good job

**Comment by ep1cman** [comment]
Thanks for your help with this feature! It will be in the next release

**Pull request merged**